### PR TITLE
[codestyle] Reduce the number of returns of this method 4, down to the maximum allowed 3

### PR DESCRIPTION
**Pull request by SrijanJha28**

### CODE REVIEW

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java [hunk 1]
```diff
@@ -283,15 +283,12 @@ public int hashCode()
     @Override
     public boolean equals(Object object)
     {
-        if (object == null) {
+        if (!(object instanceof ResourceReference)) {
```
Comment (vmassol) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: This is changing the logic and looks wrong. Where is null checked?

Comment (SrijanJha28) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: But @tmortagne had approved this in this pr : https://github.com/xwiki/xwiki-rendering/pull/40

Comment (SrijanJha28) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: and he only suggested me to do this.

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: Have you tested it by passing "null" to see if it works?

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: You can easily do that in the existing ResourceReferenceTest. Does it pass BTW? Note that it's missing a null check in the test.

Comment (SrijanJha28) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: ya it works

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: FYI I've just improved the test: https://github.com/xwiki/xwiki-rendering/commit/738ebb69402ca37cb29cb586cb0d2eeaf57fe74e

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: Just tested it too and indeed it works. "object instanceof ResourceReference" returns false when object is null.

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: I find the new code less readable than the old one, even if it's more concise. I'd personally not have changed it but let's say it's ok...

Comment (SrijanJha28) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: I told you :)

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/listener/reference/ResourceReference.java]: First rule of coding: never believe someone, always check ;)

**Pull request merged**